src/base/math/segmentedfunc.tpp

```diff
@@ -1,73 +1,31 @@
 #ifndef MATH_SEGMENTEDFUNC_TPP
 #define MATH_SEGMENTEDFUNC_TPP
 
+#include <ranges>
+
 #include "base/math/interpolation.h"
 #include "base/math/range.h"
 
 #include "segmentedfunc.h"
 
-template <typename T>
-T Math::SegmentedFunction<T>::at(double pos) const
+template <typename T, class CRTP>
+T Math::SegmentedFunction<T, CRTP>::operator()(double pos) const
 {
 	if (stops.empty()) return T();
-	if (stops.size() == 1) return stops.at(0).value;
-
-	if (pos < stops.front().pos) return stops.front().value;
-	if (pos > stops.back().pos) return stops.back().value;
```

Comment:
This was an optimization, to prevent iterating over the stop array. This usecase of having points outside the range is frequently occurs in animation easings

Reply:
I revert this `if` here: https://github.com/vizzuhq/vizzu-lib/pull/570/files#diff-6ec5da3cb41af87a90a576b0263ae1367eb23cdffb3770aca583dd4f2a62feefR76